### PR TITLE
[MU4] Accessibility. Score text editing

### PR DESCRIPTION
**Pull request by Eism**
Resolves: #10234 #10329

### CODE REVIEW

File: src/engraving/accessibility/accessibleitem.cpp [hunk 1]
```diff
@@ -165,26 +165,139 @@ QString AccessibleItem::accessibleDescription() const
...
+    textCursor->setSelectLine(startCoord.first);
+    textCursor->setSelectColumn(startCoord.second);
+
+    result = textCursor->selectedText();
+
+    delete textCursor;
+    return result;
+}
+
+int AccessibleItem::accesibleCharacterCount() const
```
Comment (igorkorsukov) [inline src/engraving/accessibility/accessibleitem.cpp:291]: missing letter `s` ;)

**Review by igorkorsukov — APPROVED** [review]

**Comment by shoogle** [comment]
Awesome! Unfortunately I get a crash when I try to edit frame text with VoiceOver enabled on macOS.

**Comment by shoogle** [comment]
Crash on macOS is fixed, thanks! Speech is working perfectly when editing frame text (title, subtitle & composer, etc.) and lyrics, but it's not working for stave text or tempo markings.

Your implementation is a bit different to what I imagined. I thought text elements would need a normal AccessibleItem to represent the element in its normal state, plus a child AccessibleTextItem to represent the item in its editing state. However, your method of adding text capabilities to normal AccessibleItems seems to work just fine!

**Comment by DmitryArefiev** [comment]
On my side on Mac (Monterey, M1), speech is not working when **entering text** (Title, Staff text, Tempo, Lyrics). 
But selection/deselection works.

**Comment by shoogle** [comment]
After the latest commits I get the opposite problem to @DmitryArefiev. Now speech works on macOS when entering text but not when selecting it or navigating through it with the arrow keys. Strangely, if I hold the Alt key while navigating then speech works better for some reason.

**Comment by Eism** [comment]
@shoogle thanks for testing!
On which text elements does the selection not work for you?

**Review by igorkorsukov — APPROVED** [review]

**Comment by DmitryArefiev** [comment]
On my side, **Mac** looks fine now - speech is working when entering text (Title, Staff text, Tempo, Lyrics) as well as selection and deselection.

**Comment by Eism** [comment]
Ok, let's merge it. 
@shoogle If this still doesn't work for you, please open a separate issue.

**Comment by shoogle** [comment]
@Eism, it turns out I accidentally enabled a VoiceOver setting called "Quick Nav" which makes the arrow keys behave very differently to how they do normally. This affected ordinary keyboard navigation within the score as well as text editing.

image

"Quick Nav" mode is toggled on and off by pressing the Right and Left arrow keys at the same time while VoiceOver is running. I must have done this accidentally at some point while testing.

Speech during text editing works perfectly when Quick Nav is disabled. Sorry for the noise!

**Pull request merged**